wagtail/admin/forms/collections.py

```diff
@@ -18,11 +18,78 @@ class Meta:
         fields = ('restriction_type', 'password', 'groups')
 
 
+class SelectWithDisabledOptions(forms.Select):
+    """
+    Subclass of Django's select widget that allows disabling options.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.disabled_values = ()
+
+    def create_option(self, name, value, *args, **kwargs):
+        option_dict = super().create_option(name, value, *args, **kwargs)
+        if value in self.disabled_values:
+            option_dict['attrs']['disabled'] = 'disabled'
+        return option_dict
+
+
+class CollectionChoiceField(forms.ModelChoiceField):
+    widget = SelectWithDisabledOptions
+
+    def __init__(self, *args, disabled_queryset=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._indentation_start_depth = 2
+        self._disabled_queryset = disabled_queryset
+        if disabled_queryset is not None:
+            self.widget.disabled_values = disabled_queryset.values_list(self.to_field_name or 'pk')
+
+    def _get_disabled_queryset(self):
+        return self._disabled_queryset
+
+    def _set_disabled_queryset(self, queryset):
+        self._disabled_queryset = queryset
+        if queryset is None:
+            self.widget.disabled_values = ()
+        else:
+            self.widget.disabled_values = queryset.values_list(self.to_field_name or 'pk', flat=True)
+
+    disabled_queryset = property(_get_disabled_queryset, _set_disabled_queryset)
+
+    def _set_queryset(self, queryset):
+        min_depth = self.queryset.aggregate(Min('depth'))['depth__min']
+        if min_depth is None:
+            self._indentation_start_depth = 2
+        else:
+            self._indentation_start_depth = min_depth + 1
+
+    def label_from_instance(self, obj):
+        return obj.get_indented_name(self._indentation_start_depth)
```

Comment:
Why? I took out the `plain_text=True` that you'd added in your code because that breaks the display in the form. 

Every single existing place where the indented form of the name is displayed, it's displayed in HTML where the `&nbsp;` are needed. That's why I change `plain_text`'s default value to False. I'm not actually sure why you even wanted a `plain_text` parameter to that method in the first place.

Reply:
Okay, understood. I wasn't 100% sure about the label value always being output in HTML, so I guess I was being a bit overcautious here. I would still prefer the default value of `plain_text` to be `True`, as there's nothing about the method in isolation that indicates it would return a HTML value. I know we're only using it here in a few places, and in all those cases, it appears that HTML _is_ indeed what we want, but to only return HTML would make the method less useful in its own right.

Reply:
Well, I mean, it *doesn't* "only return HTML". It returns plain text if you tell it to. I made it return HTML by default because every extant use of the method wants HTML. That seems like a solid reason for the default output to be HTML, but I'm no expert on API design.

Reply:
> Well, I mean, it doesn't "only return HTML". It returns plain text if you tell it to

Yes, there I was making reference to your original question:  

> I'm not actually sure why you even wanted a plain_text parameter to that method in the first place

Reply:
@coredumperror Okay, how about this? The `text_only` option becomes `html`, and the default value would be `False`? So, existing uses would changed to:

```
obj.get_indented_name(html=True)
```

That's pretty clear/concise, no?

Reply:
That seems equally as concise as the existing setup. 

I don't get why you think plaintext should be the default output, though, considering that absolutely nothing wants plaintext right now. Future callers *might* want plaintext, but basing a default on something that is totally useless today seems backwards.

Reply:
It all comes back to this point really:
>  And generally speaking, if a method/function can return a value that is useful in more contexts, and a version that is useful in fewer, you return the former by default.
Wherever possible, I try to think of the various related API changes as separate entities. They may well be reliant on each other, but there's no reason their design should tie them to each other. Someone stumbling across one element or another in the codebase will not be privy to this back-story about why a method is designed the way it is - they will not seek out this PR to view these changes as a whole like we are doing.

Reply:
Similarly, think about someone stumbling across the `label_from_instance` method above, maybe thinking about using it for some other purpose. The following shows that person a couple of quite useful things, without them having to delve any further into the code:
1. The value being returned likely includes HTML (helping them understand whether the existing code fits their needs or not)
2. If `html=True` is being specified, that probably means that the default is `False`

```
def label_from_instance(self, obj):
        return obj.get_indented_name(self._indentation_start_depth, html=True)
```

Reply:
That quote seems to fly in the face of your argument, though. The plaintext version isn't useful in *any* contexts. There is absolutely no place, anywhere, that lists the Collections in plaintext. *Every single one* is HTML, and I can't personally see any need to ever list Collections in a plaintext context.

Your argument about the readability of that API when discovered in a vacuum does make sense, but what about the reverse argument? Say we change the default, and then some time later, someone stumbles across `get_indented_name()` and sees that literally every single time it gets called, it's called with a non-default value for one of the arguments. They would wonder "Why was the code written this way? Why isn't `html=True` the default?".

I do agree that changing the argument name to `html` is probably the right thing to do, so the method is more self-documenting. But I remain unconvinced that the default should be `html=False`.

Reply:
Okay, it's late, and you seem to care about this detail more than I do. I can live with the default being `False`

Reply:
> and I can't personally see any need to ever list Collections in a plaintext context.

A CLI tool perhaps, or representation in a JSON API? Basically anything not rendered in a browser.

Reply:
Those both sound like the _perfect_ use cases for passing `html=False`. They are going to be used hundreds or thousands of times less frequently than the HTML representations will be viewed by users.